shared/actions/chat/attachment.js

```diff
@@ -182,62 +156,46 @@ function * onSelectAttachment ({payload: {input}}: Constants.SelectAttachment):
   const channelMap = ((yield call(ChatTypes.localPostFileAttachmentLocalRpcChannelMap, channelConfig, {param})): any)
 
   const uploadStart = yield Saga.takeFromChannelMap(channelMap, 'chat.1.chatUi.chatAttachmentUploadStart')
+  const messageID = uploadStart.params.placeholderMsgID
+
+  const updateLocalMessageTask = yield fork(function * () {
```

Comment:
Very good point. Considering this, I'm more inclined to go the route of adding state to the store that we look up when we receive the message, rather than waiting for the message to appear. Should be quick to change up.